docs/proposals/04-new-core-gardener-cloud-apis.md

```diff
@@ -0,0 +1,718 @@
+  # https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-aws/example/30-backupbucket.yaml#L9-L10
+  # https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-azure/example/30-backupbucket.yaml#L9-L10
+  # https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-gcp/example/30-backupbucket.yaml#L9
+  # https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-openstack/example/30-backupbucket.yaml#L9-L13
+
+---
+apiVersion: core.gardener.cloud/v1alpha1
+kind: BackupBucket
+metadata:
+  name: providerprofile1-random[:5]
+  ownerReferences:
+  - kind: ProviderProfile
+    name: providerprofile1
+spec:
+  provider:
+    type: <some-provider-name> # {aws,azure,gcp,...}
+    region: europe-central-1
+  seed: seed1
+  secretRef:
+    name: backup-operator-provider
+    namespace: backup-garden
+status:
+  lastOperation:
+    description: Backup bucket has been successfully reconciled.
+    lastUpdateTime: '2020-04-13T14:34:27Z'
+    progress: 100
+    state: Succeeded
+    type: Reconcile
+  observedGeneration: 1
+```
+
+### `BackupEntry` resource
+
+Special note: This new resource is cluster-scoped.
+
+```yaml
+# See also: https://github.com/gardener/gardener/blob/master/docs/proposals/02-backupinfra.md.
+
+apiVersion: v1
+kind: Secret
+metadata:
+  name: backup-operator-provider
+  namespace: backup-garden
+type: Opaque
+data:
+  # <some-provider-specific data keys>
+  # https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-alicloud/example/30-backupbucket.yaml#L9-L11
+  # https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-aws/example/30-backupbucket.yaml#L9-L10
+  # https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-azure/example/30-backupbucket.yaml#L9-L10
+  # https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-gcp/example/30-backupbucket.yaml#L9
+  # https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-openstack/example/30-backupbucket.yaml#L9-L13
+
+---
+apiVersion: core.gardener.cloud/v1alpha1
+kind: BackupEntry
+metadata:
+  name: shoot--core--crazy-botany--3ef42
+  namespace: garden-core
+  ownerReferences:
+  - apiVersion: core.gardener.cloud/v1alpha1
+    blockOwnerDeletion: false
+    controller: true
+    kind: Shoot
+    name: crazy-botany
+    uid: 19a9538b-5058-11e9-b5a6-5e696cab3bc8
+spec:
+  bucketName: providerprofile1-random[:5]
+  seed: seed1
+status:
+  lastOperation:
+    description: Backup entry has been successfully reconciled.
+    lastUpdateTime: '2020-04-13T14:34:27Z'
+    progress: 100
+    state: Succeeded
+    type: Reconcile
+  observedGeneration: 1
+```
+
+### `Shoot` resource
+
+Special notes:
+
+* Moved `kubeProxy` and `kubelet` configuration to worker pools.
```

Comment:
Yes, probably makes sense. We cannot move `kubeProxy` easily anyways because today it's a daemon set deployed to all nodes of all worker groups. Sure, we can split it up into one daemon set per pool, but I'd not do that now as part of this GEP but only later if we want to support something like this.

Regading the `kubelet`: Yes, makes sense, will adapt it.